lib/sql.js

```diff
@@ -156,200 +228,871 @@ SqlConnector.prototype.tableEscaped = function (model) {
+      selectStmt.merge(this.buildOrderBy(model, filter.order));
+    }
+
+    if (filter.limit || filter.skip || filter.offset) {
+      selectStmt = this.applyPagination(
+        model, selectStmt, filter);
+    }
+
+  }
+  return this.parameterize(selectStmt);
+};
+
+/**
+ * Transform the row data into a model data object
+ * @param {string} model Model name
+ * @param {object} rowData An object representing the row data from DB
+ * @returns {object} Model data object
+ */
+SqlConnector.prototype.fromRow = SqlConnector.prototype.fromDatabase = function(model, rowData) {
+  if (rowData == null) {
+    return rowData;
+  }
+  var props = this.getModelDefinition(model).properties;
+  var data = {};
+  for (var p in props) {
+    var columnName = this.column(model, p);
+    // Load properties from the row
+    var columnValue = this.fromColumnValue(props[p], rowData[columnName]);
+    if (columnValue !== undefined) {
+      data[p] = columnValue;
+    }
+  }
+  return data;
+};
+
+/**
+ * Find matching model instances by the filter
  *
  * @param {String} model The model name
- * @param {Function} callback The callback function
- * @param {Object} where The where clause
+ * @param {Object} filter The filter
+ * @param {Function} [cb] The cb function
  */
-SqlConnector.prototype.count = function count(model, callback, where) {
+SqlConnector.prototype.all = function find(model, filter, options, cb) {
   var self = this;
-
-  var whereClause = buildWhere(self, model, where);
-  this.queryOne('SELECT count(*) as cnt FROM ' +
-    this.tableEscaped(model) + ' ' + whereClause, function (err, res) {
+  // Order by id if no order is specified
+  filter = filter || {};
+  var stmt = this.buildSelect(model, filter, options);
+  this.execute(stmt.sql, stmt.params, options, function(err, data) {
     if (err) {
-      return callback(err);
+      return cb(err, []);
+    }
+
+    var objs = data.map(function(obj) {
+      return self.fromRow(model, obj);
+    });
+    if (filter && filter.include) {
+      self.getModelDefinition(model).model.include(objs, filter.include, cb);
+    } else {
+      cb(null, objs);
     }
-    callback(err, res && res.cnt);
   });
 };
+SqlConnector.prototype.findAll = SqlConnector.prototype.all;
 
 /**
- * Update attributes for a given model instance
- * @param {String} model The model name
+ * Find by id
+ * @param {String} model The Model name
  * @param {*} id The id value
- * @param {Object} data The model data instance containing all properties to be updated
+ * @param {Object} options The options object
  * @param {Function} cb The callback function
  */
-SqlConnector.prototype.updateAttributes = function updateAttributes(model, id, data, cb) {
-  if (!isIdValuePresent(id, cb)) {
+SqlConnector.prototype.findById = SqlConnector.prototype.find = function(model, id, options, cb) {
+  if (id == null) {
+    process.nextTick(function() {
+      var err = new Error('id value is required');
+      err.statusCode = 400;
```

Comment:
:+1:  for not setting HTTP-specific properties in juggler.

I think LoopBack already ensures that the `id` is always defined, thus this error should be triggered only for users calling the juggler API and/or the connector API directly.